src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/IProjectDependenciesSubTreeProvider.cs

```diff
@@ -27,19 +28,23 @@ public interface IProjectDependenciesSubTreeProvider
         event EventHandler<DependenciesChangedEventArgs> DependenciesChanged;
     }
 
-    public class DependenciesChangedEventArgs
+    public sealed class DependenciesChangedEventArgs
     {
-        public DependenciesChangedEventArgs(IProjectDependenciesSubTreeProvider provider,
-                                            string targetShortOrFullName,
-                                            IDependenciesChanges changes)
+        public DependenciesChangedEventArgs(
+            IProjectDependenciesSubTreeProvider provider,
+            string targetShortOrFullName,
+            IDependenciesChanges changes,
+            CancellationToken token)
         {
             Provider = provider;
             TargetShortOrFullName = targetShortOrFullName;
             Changes = changes;
+            Token = token;
         }
 
         public IProjectDependenciesSubTreeProvider Provider { get; }
         public string TargetShortOrFullName { get; }
         public IDependenciesChanges Changes { get; }
+        public CancellationToken Token { get; }
```

Comment:
Not thrilled with having these on eventargs, but I understand it...

Reply:
Agree but I don't know a better approach for this scenario. I don't think it's a good fit for MEF's `ImportMany`. We have Dataflow coming in to us, but then transition to an imperative/event based means to propagate data forward.